Sources/Core/AWSSDKIdentity/Sources/AWSSDKIdentity/AWSCredentialIdentityResolvers/ProfileAWSCredentialIdentityResolver.swift

```diff
@@ -172,7 +188,8 @@ public struct ProfileAWSCredentialIdentityResolver: AWSCredentialIdentityResolve
                     identityProperties: identityProperties,
                     fileBasedConfig: fileBasedConfig,
                     currentProfileName: sourceProfile,
-                    visitedProfiles: visitedProfiles
+                    visitedProfiles: visitedProfiles,
+                    credentialFeatureIDs: [CredentialFeatureID.CREDENTIALS_PROFILE_SOURCE_PROFILE.rawValue]
```

Comment:
When recursing into nested profiles via `source_profile`, we just pass a single `["o"]` into recursive calls, because duplicate credential feature IDs need to be discarded. However many levels we go into recursion, there will always be only one `o`. The moment the recursive call ends with a resolved identity from any of the leaf resolvers (Environment, IMDS (EC2), SSO, STSWebIdentity, ECS, Process, Static creds), a new feature ID will then be appended to list (e.g., `["o", "n"]` if static creds) then get bubbled back up for further additions as needed.